File: java/PROGRAMMERS/LV2_가장 큰 수.java
import java.util.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        //ArrayList<String> strNumbers = new ArrayList<String>();
        String[] strNumbers = new String[numbers.length];
        for(int i=0; i < numbers.length; i++){
            //strNumbers.add(toString(numbers[i]));
            strNumbers[i] = Integer.toString(numbers[i]);
        }
        Arrays.sort(strNumbers, new Mycomp());
        for(int i=0; i < strNumbers.length; i++){
            answer += strNumbers[i];
        }
        if(answer.startsWith("0")) return "0";
        return answer;
    }
    
    // 내림차순?
    class Mycomp implements Comparator<String> {
        public int compare(String a, String b) {
            return (b + a).compareTo(a + b);
        }
    }
}

File: cpp/PROGRAMMERS/LV2_가장 큰 수.cpp
#include <string>
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

bool compare(string a, string b) {
	
	if ((a + b).compare(b + a)) {
		cout << a + b << "," << b + a << endl;
		return (a + b) > (b + a);
	}
	return (a + b) < (b + a);
}
string solution(vector<int> numbers) {
	string answer = "";
	vector<string> strNumbers;
	for (int i = 0; i < numbers.size(); i++) {
		strNumbers.push_back(to_string(numbers[i]));
	}
	sort(strNumbers.begin(), strNumbers.end(), compare);
	for (int i = 0; i < numbers.size(); i++) {
		answer += strNumbers[i] + " " ;
	}
	if (answer[0] == '0') return "0";
	return answer;
}

int main()
{
	
	//cout << solution({ 6,10,2 }) << endl;
	//cout << solution({ 3, 30, 34, 5, 9 }) << endl;
	cout << solution({ 3, 30 }) << endl;
	/* cout << solution({ 0,0,0,1000 }) << endl; 
	cout << solution({ 110,1,100 }) << endl;
	cout << solution({ 12,121}) << endl;
	cout << solution({ 121,21}) << endl;
	cout << solution({ 21,212}) << endl;
	cout << solution({ 212,21 }) << endl;
	cout << solution({ 0,0,1000,0}) << endl;
	cout << solution({ 10000,0,0,0 }) << endl;
	cout << solution({ 9,10,100,1000 }) << endl;
	cout << solution({ 0,1,101,1001,12,123}) << endl; 
	cout << solution({ 202,1,20,11,110,11,1}) << endl;
	cout << solution({ 93,93998769}) << endl; */
}
